apps/doji-frontend/src/pages/profile/index.tsx

```diff
@@ -0,0 +1,25 @@
+import { useRouter } from 'next/router'
+import { useQuery } from 'react-query'
+
+import { httpClient } from '@frontend/services'
+
+import { IMeResponseDTO } from '@libs/api'
+
+const MyProfilePage = () => {
+  const {
+    data: user,
+    isLoading: isUserLoading,
+    isError: isUserError,
+  } = useQuery<IMeResponseDTO>('/auth/me', () => httpClient.get('/auth/me').then((res) => res.data))
+
+  if (!isUserLoading) {
+    console.log(user)
+    console.log(user.username)
+
+    const router = useRouter()
+    router.push(`/profile/${user.username}`)
+  }
```

Comment:
what is this?

Reply:
เพื่อให้เข้า /profile แล้วจะไปที่ /profile/username ของตัวเอง

Reply:
เขียนแบบนี้ ผิด `rule of hooks` ให้ไปใช้ useEffect แทน

Reply:
แล้วก็ไม่ต้อง fetch user ใหม่

ให้ใช้ `const user = useAuthStore(store => store.user)` แทน